### PR TITLE
git pull to fork from master

### PR DESCRIPTION
**Pull request by eyalpalo**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/45628)

## Description
This script was written for contributors in order to sync their branch to origin master of content repository.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Utils/git_pull_master_into_fork.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+CONTENT_URL='https://github.com/demisto/content.git'
+CURRENT=$(git branch --show-current)
```
Comment (DeanArbel) [inline Utils/git_pull_master_into_fork.sh:4]: let's add an option to provide the branch via an argument, i.e. if the user provided a branch, it'll be checked out, and if he didn't then it'll use the current branch.

File: Utils/git_pull_master_into_fork.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+{
+  git remote add upstream_content $CONTENT_URL ||
+  git remote set-url upstream_content $CONTENT_URL
+} &&
```
Comment (DeanArbel) [inline Utils/git_pull_master_into_fork.sh:9]: why did you use curly braces?

**Review by DeanArbel — CHANGES_REQUESTED** [review]
Good job. See my comments

**Comment by DeanArbel** [comment]
@yaakovi Please take a look at this PR.

File: Utils/git_pull_master_into_fork.sh (hunk 2 above)
Comment (eyalpalo) [reply Utils/git_pull_master_into_fork.sh]: changed to parentheses.

File: Utils/git_pull_master_into_fork.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+fi
+
+(
+  git remote add upstream_content $CONTENT_URL ||
+  git remote set-url upstream_content $CONTENT_URL
+) &&
+git fetch upstream_content &&
+git checkout master &&
+git rebase upstream_content/master &&
+git push -f origin master &&
```
Comment (yaakovi) [inline Utils/git_pull_master_into_fork.sh:19]: why do you force-pushing to master?

**Review by yaakovi — CHANGES_REQUESTED** [review]

File: Utils/git_pull_master_into_fork.sh (hunk 3 above)
Comment (eyalpalo) [reply Utils/git_pull_master_into_fork.sh:19]: I used the advice given [here](https://stackoverflow.com/questions/7244321/how-do-i-update-or-sync-a-forked-repository-on-github/7244456#7244456). This is relevant when the contributor already have some pushed commits in his branch (as I understand).

Comment (yaakovi) [reply Utils/git_pull_master_into_fork.sh:19]: add a comment at the top of the file that only contributors should run this script.

**Review by yaakovi — APPROVED** [review]

**Review by DeanArbel — APPROVED** [review]
Good job. Make sure to add this to our contributor's docs, and update it in our confluence `contributions` page.

**Pull request merged**